src/main/scala/zio/prelude/Hash.scala

```diff
@@ -231,6 +232,17 @@ object Hash extends Lawful[Hash] {
   implicit def SetHash[A]: Hash[Set[A]] =
     default
 
+  /**
+   * Derives a `Hash[SortedMap[A, B]]` given a `Hash[A]` and `Hash[B]`.
+   */
+  implicit def SortedMapHash[A: Hash, B: Hash]: Hash[SortedMap[A, B]] =
```

Comment:
Does relying on the implicit `Hash[A]` make sense? The `SortedMap` is going to be constructed with an arbitrary ordering and Scala's definition of equality, so do we have a guarantee that those are going to be consistent with the definition of equality embodied in the `Hash` instance?